### ISSUE TITLE
zlib-ng now unusable with ElectricFence

### ISSUE THREAD

**Issue by lanurmi** [zlib-ng/zlib-ng#1809]
Zlib-ng has become unusable with ElectricFence. I encountered this on Fedora Rawhide, when running something utilizing zlib-ng. Luckily, the same problem can be seen with the `gh1235` test:

```
$ ef ./test/gh1235

  Electric Fence 2.2.2 Copyright (C) 1987-1999 Bruce Perens <bruce@perens.com>
bytes =  0, deflateBound = 23, total_out = 23

ElectricFence Aborting: free(1589880): address not from malloc().
/usr/bin/ef: line 20: 2800494 Illegal instruction     (core dumped) ( export LD_PRELOAD=libefence.so.0.0; exec "$@" )
```

Git-bisecting reveals that this happens since 42db8b86a842e92e2dd9695860b03bcbdf3fe1c0.

Whether it is zlib-ng's or ElectricFence's fault is a question can be debated.

**Comment by tuliom** [comment]
I am afraid this is actually a bug in ElectricFence.
I found an implementation [here](https://github.com/kallisti5/ElectricFence/blob/master/src/efence.c) although I'm not sure if this is the right repository.

zlib-ng is now using `aligned_alloc()`, which is a function that got added to C11.
If you see the code above, ElectricFence is interposing other functions (`malloc`, `free`, `calloc`, etc.) but it does not implement `aligned_alloc`.

If my interpretation is correct, that means ElectricFence is not tracking calls to `aligned_alloc` and errors out when `free` is called with a pointer allocated via `aligned_alloc` from the system C library.

**Comment by mtl1979** [comment]
ElectricFence is abandonware, it's not maintained anymore. The person who forked the original sources works on same project nowadays as I do...

**Comment by lanurmi** [comment]
Agreed, it is a bug (or rather missing functionality) in ElectricFence. I'm also aware that it is not maintained, but I've submittted [a bug report to Fedora](https://bugzilla.redhat.com/show_bug.cgi?id=2326420), along with a patch to fix the issue. (Just FYI, in case someone hits the some issue and is looking for a solution.)

**Comment by nmoinvaz** [comment]
Closing. It was reported on ElectricFence's GitHub issues.
